src/ForwardCompatibility/Driver/ResultStatement.php

```diff
@@ -1,59 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Doctrine\DBAL\ForwardCompatibility\Driver;
-
-use Doctrine\DBAL\Driver\DriverException;
-use Doctrine\DBAL\Driver\ResultStatement as BaseResultStatement;
-
-/**
- * Forward compatibility extension for the ResultStatement interface.
- */
-interface ResultStatement extends BaseResultStatement
-{
```

Comment:
Should we maybe mark this as internal in lower branches? Since we remove it here, we don't want people to type hint against it. If it's ok to type hint against it, maybe we should keep it, deprecate it, and empty it.

Reply:
Let's not mark it internal so that people don't feel wrong about using it, and let's empty and deprecate it instead of removal in `3.0.x`.

Most likely, this interface will go away or get renamed/repurposed before `2.11.x` is released but I'm fine with keeping it always releasable.